src/System.Management.Automation/engine/remoting/commands/EnterPSHostProcessCommand.cs

```diff
@@ -602,9 +602,16 @@ private static int[] GetProcIdsFromNames(string[] names)
                         continue;
                     }
 
-                    if (namePattern.IsMatch(proc.ProcessName))
+                    try
                     {
-                        returnIds.Add(proc.Id);
+                        if (namePattern.IsMatch(proc.ProcessName))
+                        {
+                            returnIds.Add(proc.Id);
+                        }
+                    }
+                    catch (InvalidOperationException)
```

Comment:
Now we have no need to check proc.HasExited :-)

Reply:
No, it is better to not rely on catching thrown exception when possible.  The try/catch is only for race conditions.